### PR TITLE
numbat: init at 1.6.3

### PR DESCRIPTION
**Pull request by giomf**
## Description of changes
Hi,

I want to contribute my first package called [numbat](https://numbat.dev/) which is a _High precision scientific calculator with full support for physical units._
Since it is my first package please point me into the right direction if I did something wrong.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by h7x4** [comment]
Hello @giomf, welcome to nixpkgs!

We have a few contributing guidelines as you might've seen, in particular I'd like to point you to the [commit conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#commit-conventions). You'll have to split up your commit into two - one that adds you to the maintainer list and another which is the init numbat commit.

**Comment by giomf** [comment]
Yes I already saw this but I have a question regarding the maintainer commit:
How to prefix it since its not related to any package?
From the docs: `(pkg-name | nixos/<module>): (from -> to | init at version | refactor | etc)`

File: pkgs/by-name/nu/numbat/package.nix
Comment (h7x4) [inline pkgs/by-name/nu/numbat/package.nix:32]: I belive the convention is that `platform` should be set to what upstream specifies, or what we think is possible if nothing is specified. In this case, this program can probably run on a lot more that just `linux`. `rustPlatform` sets a default set of architectures which matches what rust can compile to, so I think this field can just be dropped in this case.

```suggestion
```

File: pkgs/by-name/nu/numbat/package.nix
Comment (h7x4) [inline pkgs/by-name/nu/numbat/package.nix:19]: Could you explain why this is set?

**Review by h7x4 — COMMENTED** [review]
Not much to comment on here, neat!

File: pkgs/by-name/nu/numbat/package.nix
Comment (giomf) [reply pkgs/by-name/nu/numbat/package.nix:19]: To be honest no. 
As far Im aware of numbat has no features to activate anyway. So we could remove it I guess?

Comment (h7x4) [reply pkgs/by-name/nu/numbat/package.nix]: Sounds good!

**Comment by natsukium** [comment]
@ofborg eval

File: pkgs/by-name/nu/numbat/package.nix
Comment (h7x4) [inline pkgs/by-name/nu/numbat/package.nix:26]: :)

```suggestion
    changelog = "https://github.com/sharkdp/numbat/releases/tag/v${version}";
```

**Review by h7x4 — APPROVED** [review]
Built and tested application, works nicely!

`githubId` checks out: https://api.github.com/user/35076723

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

Result of `nixpkgs-review pr 264241` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>numbat</li>
  </ul>
</details>

Marking as approved now, with the assumption that `meta.changelog` gets fixed

File: pkgs/by-name/nu/numbat/package.nix
Comment (giomf) [reply pkgs/by-name/nu/numbat/package.nix]: Good catch!

**Comment by giomf** [comment]
@h7x4 have I missed something or why does it takes so long to complete the checks?

**Comment by h7x4** [comment]
It's just the aarch64-darwin (read: ARM-based Apple mac, M1/M2 chip) builder left. We don't really have a lot of machines which can run these tests, so they usually take a really long time (if they even complete at all). The fact that the tests run fine on x86_64 darwin, x86_64 linux and aarch64-linux is a good sign, and since it's not a very platform specific package, it will *probably* work on aarch64-darwin

**Comment by giomf** [comment]
Ok, thanks for the clarification.
So I will have to wait before I can merge...

**Comment by h7x4** [comment]
I don't think so. Not necessarily at least. You could ask someone with an ARM mac to verify that the package works correctly if you suspect it won't, but I think this is ready for merge despite the queued test. You'll just have to wait for someone with commit rights to have a look at it.

I'm not sure how familiar you are with the process of getting things merged in nixpkgs, considering this is your first PR. It can be a bit of a slow process, because we have so many PRs compared to the number of reviewers and committers. There are ways you can try to ask for more reviews, namely through the "PRs ready for *" threads on discourse, or in our matrix dev channels.

Alternatively, if you feel like you're up to the challenge, we would be really happy to have some more reviewers. There are several ways to help out with this that doesn't require deep internal knowledge of nix, NixOS or nixpkgs. Something like just building and testing out the software (using tools like [nixpkgs-review](https://github.com/Mic92/nixpkgs-review)) in the different PRs would be a great help. This would speed up the process for everyone.

**Review by h7x4 — APPROVED** [review]

**Comment by giomf** [comment]
> I'm not sure how familiar you are with the process of getting things merged in nixpkgs, considering this is your first PR. It can be a bit of a slow process, because we have so many PRs compared to the number of reviewers and committers. There are ways you can try to ask for more reviews, namely through the "PRs ready for *" threads on discourse, or in our matrix dev channels.

Thanks for the hint. I will open a thread.

> Alternatively, if you feel like you're up to the challenge, we would be really happy to have some more reviewers. There are several ways to help out with this that doesn't require deep internal knowledge of nix, NixOS or nixpkgs. Something like just building and testing out the software (using tools like [nixpkgs-review](https://github.com/Mic92/nixpkgs-review)) in the different PRs would be a great help. This would speed up the process for everyone.

This sounds good. I would love to help where I can.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 264241` at d074b63c run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/60c0815fa64239e1993832401bcd1b78">numbat</a></li>
  </ul>
</details>

File: pkgs/by-name/nu/numbat/package.nix
Comment (natsukium) [inline pkgs/by-name/nu/numbat/package.nix:3]:
```suggestion
, rustPlatform
, darwin
```

File: pkgs/by-name/nu/numbat/package.nix
Comment (natsukium) [inline pkgs/by-name/nu/numbat/package.nix:18]:
```suggestion

  buildInputs = lib.optionals stdenv.isDarwin [
    darwin.apple_sdk.frameworks.Security
  ];

```

**Comment by natsukium** [comment]
Whenever you need a review or merge, please contact me.
I may not always be able to do it, but I'm sure I can help.

File: pkgs/by-name/nu/numbat/package.nix
Comment (giomf) [reply pkgs/by-name/nu/numbat/package.nix:19]: Why is  `darwin.apple_sdk.frameworks.Security` needed? Or why do I only need the `Security` part of the`apple_sdk`.

Comment (natsukium) [reply pkgs/by-name/nu/numbat/package.nix:20]: Unfortunately I do not know where to find the document.
I just added it because the compiler reported that Security was missing.

Since apple_sdk.frameworks originally exists in the darwin environment, many packages implicitly depend on it, such as this package.
However, these are impure and need to be explicitly added to the input in nix.
The frameworks are listed in [this file](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/darwin/apple-sdk/frameworks.nix), but it is too much to include them all, so only the necessary ones should be added.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 264241` at 53da5fe9 run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>numbat</li>
  </ul>
</details>

File: pkgs/by-name/nu/numbat/package.nix
Comment (giomf) [reply pkgs/by-name/nu/numbat/package.nix:20]: Yes I also found this file but it was a little overwhelming.
Thanks for clarification though.

**Review by h7x4 — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**
